extensions/netlify/src/components/deploys.tsx

```diff
@@ -0,0 +1,242 @@
+import { Action, ActionPanel, Icon, List } from '@raycast/api';
+import { useEffect, useState } from 'react';
+
+import api from '../utils/api';
+import { OpenRepo } from '../components/actions';
+import {
+  formatDate,
+  getDeployUrl,
+  getStatusText,
+  handleNetworkError,
+} from '../utils/helpers';
+import { getStatusIcon } from '../utils/icons';
+import { Deploy } from '../utils/interfaces';
+
+interface Props {
+  siteId: string;
+  siteName: string;
+}
+
+export default function DeployListView(props: Props) {
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [deploys, setDeploys] = useState<Deploy[]>([]);
+
+  const [query, setQuery] = useState<string>('');
+  const [context, setContext] = useState<string>('');
+
+  const { siteId, siteName } = props;
+
+  useEffect(() => {
+    async function fetchDeploys() {
+      try {
+        const deploys = await api.getDeploys(siteId);
+        setDeploys(deploys);
+        setLoading(false);
+      } catch (e) {
+        setLoading(false);
+        handleNetworkError(e);
+      }
+    }
+
+    fetchDeploys();
+  }, []);
```

Comment:
Note that you can use some hooks from `@raycast/utils` to make async stuff easier to manage: either [useFetch](https://developers.raycast.com/utilities/react-hooks/usefetch) or [useCachedPromise](https://developers.raycast.com/utilities/react-hooks/usecachedpromise). In your case, `useFetch` should be enough.

Reply:
How can `useFetch` be used if authorization headers need to be sent with the request?

Reply:
`useFetch` options extends [RequestInit](https://github.com/nodejs/undici/blob/v5.7.0/types/fetch.d.ts#L103-L117) so you should be able to send authorization headers like the example below:

```ts
const { isLoading, data, error } = useFetch<Advice>(url, {
  headers: {
    // ...
  },
  // ...
});
```

Reply:
Thanks for the suggestion @thomaslombart! I ended up going with `usePromise`, and was able to delete a lot of redundant code. All of the `usePromise` changes are contained in https://github.com/raycast/extensions/pull/4270/commits/7bcde176a05b40b6ddb8bcf63a2d607ca8725610.